pkgs/applications/audio/reaper/default.nix

```diff
@@ -1,10 +1,37 @@
-{ lib, stdenv, fetchurl, autoPatchelfHook, makeWrapper
-, alsaLib, xorg, libjack2
-, gtk3, pango, gdk-pixbuf, cairo, glib, freetype
-, libpulseaudio, xdg_utils
+{ config, lib, stdenv
+, fetchurl
+, autoPatchelfHook
+, makeWrapper
+
+, alsaLib
+, gtk3
+, lame
+, ffmpeg
+, vlc
+
+, jackSupport ? true, libjack2
+, pulseaudioSupport ? config.pulseaudio or true, libpulseaudio
 }:
 
-stdenv.mkDerivation rec {
+let
+  # Dynamic loading of plugin dependencies does not adhere to rpath of
+  # reaper executable that gets modified with runtimeDependencies.
+  # Patching each plugin with DT_NEEDED is cumbersome and requires
+  # hardcoding of API versions of each dependency.
+  # Setting the rpath of the plugin shared object files does not
+  # seem to have an effect for some plugins.
+  # We opt for wrapping the executable with LD_LIBRARY_PATH prefix.
+  pluginDeps = [
+    lame
+    ffmpeg
+    vlc
+  ];
+  wrapCommand = lib.optionalString (builtins.length pluginDeps != 0) ''
```

Comment:
I think we inline the wrapCommand. It should make it easier for other people to follow the build commands.

Reply:
Makes sense, thanks!